lamindb/dev/db/_add.py

```diff
@@ -81,28 +81,84 @@ def add(  # type: ignore
             return results
         else:
             records = [model(**fields)]
-    for record in records:
-        if isinstance(record, DObject) and hasattr(record, "_local_filepath"):
-            upload_data_object(record, use_fsspec=use_fsspec)
+
     if session is None:  # assume global session
         session = settings.instance.session()
         settings.instance._cloud_sqlite_locker.lock()
         close = True
     else:
         close = False
+
+    added_records = []
+    raise_error = False
     for record in records:
-        session.add(record)
-    session.commit()
-    for record in records:
+        # commit metadata to database
+        prepare_filekey_metadata(record)
+        try:
+            session.add(record)
```

Comment:
This will always go well and if it doesn't, we want a proper traceback as we'll deal with a fundamental problem - the "expected risk" that we want to mitigate by manually adding try except lies in the db transaction (say, unstable internet connection, overloaded network, etc.)

Reply:
Implemented. See [here](https://github.com/laminlabs/lamindb/pull/476#discussion_r1097623303).